x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ExpressionBuilder.java

```diff
@@ -223,6 +223,11 @@ public UnresolvedAttribute visitQualifiedName(EsqlBaseParser.QualifiedNameContex
         return new UnresolvedAttribute(source(ctx), Strings.collectionToDelimitedString(strings, "."));
     }
 
+    @Override
+    public List<Attribute> visitQualifiedNames(EsqlBaseParser.QualifiedNamesContext ctx) {
+        return visitList(this, ctx.qualifiedName(), Attribute.class);
+    }
```

Comment:
Do we need this? Is there some other rule I should be using instead?

Reply:
You shouldn't - the default implementation should automatically do the right thing. I assume you tried it though and it failed - what was the error?

Reply:
They come back the wrong type. I'm wondering if there's a different `visit` I should be using - it feels strange that I'm the first one to want names in the shape that I want them. I'll bet there's something else similar and I just didn't see it.